MD/md-20/README.md

```diff
@@ -0,0 +1,93 @@
+- **L1 ERC20 $MOVE**: The ERC20 Move token on Ethereum (L1).
+- **Pool**: A holding place for assets or state data in the contract on either L1 or L2.
+
+## Desiderata
+
+### 1. Seamless Acquisition of L2 $MOVE on Completion of Transfer
+
+- **User Journey**: After a transfer is initiated and the secrets are revealed, the Bridge Relayer calls `complete_bridge_transfer`. At this point, the bridge move module mints the `BRIDGE-FA`, a temporary holding asset used to swap for L2 $MOVE.
+- **Justification**: Using the BRIDGE-FA provides a clean API and audited safety features. Furthermore, it has already been implemented. The prior implementation was called `MOVETH`, but the source code is the same. [MOVETH.move source code](https://github.com/movementlabsxyz/movement/blob/main/protocol-units/bridge/move-modules/sources/MOVETH.move). The [atomic_bridge_counterparty.move](https://github.com/movementlabsxyz/movement/blob/main/protocol-units/bridge/move-modules/sources/atomic_bridge_counterparty.move) module makes calls to the FA module.
+
+### 2. Safe and Secure Access to L2 $MOVE Pool
+
+- **Justification**: The Bridge Relayer keys should not have access to the $MOVE pool. If they did, they could arbitrarily mint L2 $MOVE, effectively offsetting the balance between L1 ERC20 $MOVE and L2 $MOVE, potentially incurring huge losses for Movement Labs.
+
+### 3. Gas Sponsorship for the L2 Swap
+
+- **User Journey**: After a transfer from L1 to L2 is completed, the user expects to receive L2 $MOVE equivalent to the ERC20 $MOVE that was transferred into their Movement Wallet.
+- **Justification**: At the initial step of the swap (when the transfer is completed on Movement L2), the user requires some L2 $MOVE to execute the swap transaction.
+
+### 4. Fallback Mechanism for Failed Swaps
+
+- **User Journey**: The user is waiting for their transfer to complete. There is an AWS outage, and our relayer goes down. This happens after the transfer is completed, but the Swap TX was never called.
+- **Justification**: The swap could fail for several reasons, such as connectivity or liveness issues. In such cases, a suitable mechanism should be implemented to retry or refund the swap.
+
+## Guidance and Suggestions
+
+At the earliest stage of the L1 to L2 transfer, the Bridge Relayer service should transfer some of its owned L2 $MOVE to the user associated with the recipient on the `bridgeTransferId`, **IF** the user does not already possess L2 $MOVE. This is the sponsored transaction, and it should be a very nominal amount.
+
+The safest and most secure way to implement the swap step of the Transfer sequence is to maintain the pattern already implemented in the atomic bridge contracts. Namely, the bridge counterparty module mints an `FA` and gives it to the user.
```

Comment:
can we not make the Aptos Coin module FA?

Reply:
No, it's not possible in the current version of aptos-core. There _is_ however code for this that @andyjsbell has found, but it's been feature flagged out and has not hit mainnet. We decided earlier on to not go down this path to keep parity with aptos-core mainnet now.